### ISSUE TITLE
vina error

### ISSUE THREAD

**Issue by ever4cyth** [oddt/oddt#5]
Try to see if my installation works using script in your paper, but autodockvina gives error. Is the parameters passed to vina normal? Thanks!

from oddt.virtualscreening import virtualscreening as vs
# Initiate virtual screening pipeline using 4 CPUs
pipeline = vs(n_cpu=4)
# Load ligands from a mol2 file
pipeline.load_ligands('mol2', 'ampc/actives_final.mol2.gz')
# Filter ligands by weight and solubility
pipeline.apply_filter('150 < mol.molwt < 350')
pipeline.apply_filter('0 < mol.logp < 5')
# Dock entire library to receptor, autocenter docking box on ligand
pipeline.dock('autodock_vina', 'ampc/receptor.pdbqt', 'ampc/crystal_ligand.mol2')
# Write docked ligands conformations to a file (including scores)
pipeline.write('mol2', 'ampc_docked.mol2', overwrite=True, opt={'c':None})
# Rescore docking results using RFscore NNscore
pipeline.score(function='rfscore', protein='ampc/receptor.pdbqt')
pipeline.score(function='nnscore', protein='ampc/receptor.pdbqt')
# Write computed scores to csv
pipeline.write_csv('ampc_docked.csv')

subprocess.CalledProcessError: Command '['/home/magnolia/protein/autovina/bin/vina', '--receptor', 'ampc/receptor.pdbqt', '--ligand', '/tmp/autodock_vina_RbPIq2/ligands_ybJzTA/0_308.pdbqt', '--out', '/tmp/autodock_vina_RbPIq2/ligands_ybJzTA/0_308_out.pdbqt', '--center_x', '0', '--center_y', '0', '--center_z', '0', '--size_x', 'a', '--size_y', 'm', '--size_z', 'p', '--cpu', '1', '--exhaustiveness', '8', '--num_modes', '9', '--energy_range', '3']' returned non-zero exit status 1

**Comment by mwojcikowski** [comment]
I must look into that deeper, although there seam to be an issue with implicit arguments order. The code was fully functioning and tested. For now try changing dock method call to explicitly give auto_ligand parameter:

> pipeline.dock('autodock_vina', 'ampc/receptor.pdbqt', auto_ligand='ampc/crystal_ligand.mol2')

**Comment by mwojcikowski** [comment]
Please upgrade ODDT from pip. The bug is fixed.
> pip install -U --no-deps oddt

**Comment by ever4cyth** [comment]
Thanks! it works now.

**Comment by SuhadJihad** [comment]
Hi,
Thanks for your code,
When I try to use it every thing is OK but when I reach  pipeline.write('mol2', 'ampc_docked.mol2', overwrite=True, opt={'c':None}) nothing appear even the prompt the terminal of python is latch like you see in the attachment 
![screenshot from 2015-11-14 01 29 52](https://cloud.githubusercontent.com/assets/15838970/11159449/aa1e01cc-8a70-11e5-9f37-59a109bb1ccc.png)
and when I interrupt it I get this error:
![screenshot from 2015-11-14 01 41 13](https://cloud.githubusercontent.com/assets/15838970/11159473/e944e032-8a70-11e5-943d-36da5d83b388.png)
I appreciate any help ....

**Comment by SuhadJihad** [comment]
Sorry, but I forget to tell you that the software is up-to-date that you can see here
![screenshot from 2015-11-14 02 08 32](https://cloud.githubusercontent.com/assets/15838970/11159893/d1f7f352-8a74-11e5-8189-055c0d07d140.png)

**Comment by mwojcikowski** [comment]
Pipelines are lazy. This means that unless you execute the chained pipeline by fetching the ligands or writing to the output file actually no computation is done.

So in your case the docking is being done when you enter pipeline.write(...) command. Be patient and check the top/htop for the running processes.

**Comment by SuhadJihad** [comment]
Thank you for your answer ,you are right I wait and finally get a result but I face another problem :
![screenshot from 2015-11-17 00 54 00](https://cloud.githubusercontent.com/assets/15838970/11196022/4ebd3ae8-8cc6-11e5-8c63-10be5d2ca955.png)
When I read the journal and the code I understand that there is no pickle files but I am newbie in this field and couldn't resolve it.
![screenshot from 2015-11-17 01 03 32](https://cloud.githubusercontent.com/assets/15838970/11196174/1377b39a-8cc7-11e5-864e-9992651c34ca.png)
Thanks a lot for explanation.

**Comment by mwojcikowski** [comment]
Pickle is a Python object saved to a file for later use. It is not recommended to use pickle files from unknown sources because of security reasons. Other than that all the libraries version must match.

Considering this scoring functions in ODDT have to be built upon first run. The RF-score is really fast, but NNscore can be veeery slow (it might take over an hour to build). According to original release of NNScore 2.0 there are 1000 neural-netwoks being built, then the best are selected.

To sum up - just be patient once again.

PS.
We are working hard to do something about it, like switching to netter neural-network library, although it's not that  straightforward.

**Comment by SuhadJihad** [comment]
Thanks for your quick response,
when I wait nnscore an error appear and nothing happened because I check the htop.
![screenshot from 2015-11-18 03 40 54](https://cloud.githubusercontent.com/assets/15838970/11229693/358046cc-8da9-11e5-8ef6-f7163b33d1ed.png)
and when I try to write to the _csv file the file exist but it is 0 byte.
then I try to score the result only with rfscore I get this error when I want to write the result to _csv file
![screenshot from 2015-11-18 04 06 12](https://cloud.githubusercontent.com/assets/15838970/11229770/fa89a3dc-8da9-11e5-9323-b7d7f2011785.png)
and this error appear many times:
AttributeError: Molecule has no such property: atom_dict

Then I repeat the scoring by these two function this time have no error
![screenshot from 2015-11-18 04 17 36](https://cloud.githubusercontent.com/assets/15838970/11229909/54b83548-8dab-11e5-93b0-dd5edbe2cc94.png)
but when I start writing nothing happened and 0byte file exist
![screenshot from 2015-11-18 04 19 14](https://cloud.githubusercontent.com/assets/15838970/11229943/8f17bc72-8dab-11e5-862b-70b9c11d54d4.png)

**Comment by mwojcikowski** [comment]
First of all please move that question to the mailing list [oddt@googlegroups.com], since that has nothing to do with current **closed** issue.

Secondly, as you can see the NNScore has trained itself. There should be .pickle file in working directory.

Third thing - the Molecule class is somehow loaded from outside of ODDT, since it has some attributes missing (most probable from pybel/cinfony). Have you tried clean terminal or assembling a script and executing the pipeline separately?
